core/src/main/java/com/linecorp/armeria/server/file/AbstractHttpFile.java

```diff
@@ -197,7 +200,15 @@ public final CompletableFuture<HttpResponse> read(Executor fileReadExecutor, Byt
         requireNonNull(alloc, "alloc");
 
         return readAttributes(fileReadExecutor)
-                .thenApply(attrs -> read(fileReadExecutor, alloc, attrs))
+                .thenApply(attrs -> {
+                    final HttpResponse res = read(fileReadExecutor, alloc, attrs);
+                    if (res != null) {
+                        return res;
+                    }
+
+                    logger.debug("{} is not found. attrs: {}", pathOrUri(), attrs);
```

Comment:
I think we use `warn` instead `debug` because the user added this `HttpFile` that doesn't exist.